engine/src/cmd/script/script_statement.cpp

```diff
@@ -77,7 +77,10 @@ void Mission::doIf( missionNode *node, int mode )
 
         int nr_subnodes = node->subnodes.size();
         if (nr_subnodes != 3) {
-            fatalError( node, mode, "an if-statement needs exact three subnodes, not "+nr_subnodes );
+            char buffer[2048];
+            memset(buffer, 0, sizeof(char)*2048);
+            snprintf(buffer, sizeof(char)*2047, "an if-statement needs exact three subnodes, not %d", nr_subnodes );
+            fatalError( node, mode, buffer);
```

Comment:
How about just using `boost::format(...).str()`? Something along the lines of:

```suggestion
            std::string error_message = (boost::format("an if-statement needs exactly three subnodes, not %1%") % nr_subnodes).str();
            fatalError( node, mode, error_message.c_str() );
```

Reply:
Although what you have here probably works too. It's just that Windows / Visual Studio might complain about it.

Reply:
oh yeah - MSVC will probably complain about it since I didn't use the `_s` versions. Happy to use your suggestion.